### DISCUSSION TITLE
Panels on Dashboard not showing after readd

### DISCUSSION THREAD

**Discussion by BumBum94** [primefaces/community#683] [PrimeFaces] [answered]
Hello
In our application we use the dashboard for various online panels such as “Bookings” or “Charts”.
Each user can edit and save their dashboard individually.
Everything works the way we want it to.
![dashboard](https://github.com/primefaces/community/assets/29804746/84c73502-f017-477e-8e49-6e54ad9bde25)

There is only one problem when I remove a panel from the dashboard with the onClose event and then want to add the same panel again, the panel is not displayed.
`public void onClose(CloseEvent event) {
		String panel = event.getComponent().getId().toLowerCase();
		for (DashboardWidget dc : model.getWidgets()) {
			dc.removeWidget(event.getComponent().getId());
		}
		ZMIBoardPanel p = getTerminalDAO().getBoardPanelByName(panel);
		panels.add(p);
		bubbleSort(panels);
	}`
`public void addPanel() {
		
		initPanel(newPanel.toLowerCase());
		model.addWidget(new DefaultDashboardWidget(newPanel.toLowerCase(), RESPONSIVE_CLASS));
		// model.getColumns().forEach(t -> t.getWidgets().forEach(w -> w.)
		delPanelFromList(newPanel);
	}`

 Only after reloading the page does it appear as it should.
If you examine the page in the browser you can see that the panel is added but is not displayed because the CSS class "ui-helper-hidden" is assigned to it.
If you remove this, the panel will be displayed but without any content.
![ui-helper-hidden](https://github.com/primefaces/community/assets/29804746/b88c39f0-d9d6-4631-a66e-94b730d29c37)

Can anyone Help?

**Comment by melloware** [comment]
Please provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.

**Comment by BumBum94** [comment, marked as answer]
i solved the problem by reloading the page afte the close event 
`<p:ajax event="close" listener="#{terminalBean.onClose}"
              update="dash :panelDialog:panel"
              onstart="PF('statusDialog').show();"
	      oncomplete="window.location.reload();" />`
